tournamentcontrol/competition/_mediaupload.py

```diff
@@ -0,0 +1,94 @@
+"""
+Custom media upload classes for YouTube thumbnail management.
+
+This module provides database-sourced media upload classes that are compatible
+with the Google API client library, similar to MediaFileUpload but reading
+data from Django model fields instead of files.
+"""
+
+import io
+from googleapiclient.http import HttpRequest, MediaUpload
+
+
+class MediaDatabaseUpload(MediaUpload):
```

Comment:
There is no longer a strict association with the database in this implementation, rename to be more representative.

Update documentation, tests, and references accordingly.